types/handler.ts

```diff
@@ -0,0 +1,51 @@
+import { JsonRpcProvider } from "@ethersproject/providers";
+import { ChainId, networkCurrencies, networkExplorers, networkIds, networkNames, networkRpcs, tokens } from "./constants";
+
+export type ValidBlockData = {
+  jsonrpc: string;
+  id: number;
+  result: {
+    number: string;
+    timestamp: string;
+    hash: string;
+  };
+};
+
+export type Token = {
+  decimals: number;
+  address: string;
+};
+
+export type NativeToken = {
+  symbol: string;
+  decimals: number;
+};
+
+export type HandlerInterface = {
+  getProvider(): JsonRpcProvider | undefined;
+  clearInstance(): void;
+  getFastestRpcProvider(): Promise<JsonRpcProvider | undefined>;
+  testRpcPerformance(): Promise<JsonRpcProvider | undefined>;
+};
+
+export type HandlerConstructorConfig = {
+  networkId: number;
+  networkName?: string;
+  networkRpcs?: string[];
```

Comment:
Optional and array of strings is gonna be a mess to check. Probably better to just initialize as an empty array and check if it's an empty array etc